Add JWTManager for authentication with tests

almogalf opened the pull request:
# Added `JWTManager` for User Authentication (Including Tests & Config Updates)

## Summary
This PR introduces the `JWTManager` class, responsible for handling user authentication via JSON Web Tokens (JWT).

## Features Implemented:
- **Token Generation:** Secure JWT token creation for authentication.
- **Token Verification:** Ensures token validity and expiration handling.
- **User Authentication:** Extracts and verifies user identity from the JWT payload.

## Additional Changes:
- **Added Unit Tests:** Implemented `tests/test_jwt_manager.py` to validate JWT functionalities.
- **Updated `config.py`:** Added JWT-related configurations (`JWT_SECRET_KEY`, `JWT_ALGORITHM`, and expiration settings).

## Next Steps:
- Integrate `JWTManager` with the authentication system.

Please review and provide feedback.

almogalf committed:
Add JWTManager for authentication with token handling. Includes unit tests for token generation, verification, and expiration. Updated config.py with JWT secret key and expiration settings.

TamarAdn commented on app/models/jwt_manager.py:
The code is clear and uses descriptive names

TamarAdn commented on app/models/jwt_manager.py:
The error handling in verify_token and refresh_access_token is straightforward, catching specific JWT errors and wrapping them in a custom AuthenticationError. This is very good!

TamarAdn reviewed (changes requested):
Both generate_access_token and generate_refresh_token share nearly identical logic. Refactoring them into a common private method (like _generate_token) can reduce duplication and simplify future updates.  Maybe tests should be modified accordingly.

almogalf committed:
Refactored JWT token generation to reduce duplication

- Introduced  as a common method for access and refresh token creation.
- Updated  and  to use .
- Ensured consistency in expiration handling.
- Modified tests if necessary to align with the refactored structure.

almogalf committed:
Refactor JWTManager: Improved token generation and error handling

TamarAdn commented on app/models/jwt_manager.py:
```diff
@@ -0,0 +1,157 @@
+    """Manages JWT token operations - generation, verification, and refreshing."""
+
+    @staticmethod
+    def _generate_token(
```
Thank you for making the changes!
